libp2p/Host.cpp

```diff
@@ -547,46 +547,42 @@ void Host::requirePeer(NodeID const& _n, NodeIPEndpoint const& _endpoint)
         return;
     }
 
+    if (_n == id())
     {
-        Guard l(x_requiredPeers);
-        m_requiredPeers.insert(_n);
+        cnetdetails << "Ignoring the request to connect to self " << _n;
+        return;
     }
 
-    if (_n == id())
+    if (!_n)
     {
-        cnetdetails << "Ignoring the request to connect to self " << _n;
+        cnetdetails << "Ignoring the request to connect to null node id.";
         return;
     }
 
-    Node node(_n, _endpoint, PeerType::Required);
-    if (_n)
     {
-        // create or update m_peers entry
-        shared_ptr<Peer> p;
-        DEV_RECURSIVE_GUARDED(x_sessions)
-            if (m_peers.count(_n))
-            {
-                p = m_peers[_n];
-                p->endpoint = node.endpoint;
-                p->peerType = PeerType::Required;
-            }
-            else
-            {
-                p = make_shared<Peer>(node);
-                m_peers[_n] = p;
-            }
-        // required for discovery
-        addNodeToNodeTable(*p, NodeTable::NodeRelation::Unknown);
+        Guard l(x_requiredPeers);
+        m_requiredPeers.insert(_n);
     }
-    else
+
+    Node const node(_n, _endpoint, PeerType::Required);
+    // create or update m_peers entry
+    shared_ptr<Peer> p;
+    DEV_RECURSIVE_GUARDED(x_sessions)
     {
-        if (!addNodeToNodeTable(node))
-            return;
-        scheduleExecution(600, [this, _n]() {
-            if (auto n = nodeFromNodeTable(_n))
-                requirePeer(n.id, n.endpoint);
-        });
+        if (contains(m_peers, _n))
+        {
+            p = m_peers[_n];
```

Comment:
Sorry for not clearly describing the issue. If `_n` is found the iterator can be used here.

Reply:
ah right

Reply:
fixed